features/support/api_helpers.rb

```diff
@@ -98,7 +98,15 @@ def set_supplier_on_framework(framework_slug, supplier_id, status)
     frameworkInterest: { onFramework: status },
     updated_by: "functional tests",
   })
-  expect(response.code).to eq(200), _error(response, "Failed to update agreement status #{supplier_id} #{framework_slug}")
+  expect(response.code).to eq(200), _error(response, "Failed to update supplier_framework status #{supplier_id} #{framework_slug}")
+end
+
+def confirm_company_details_for_framework(framework_slug, supplier_id)
+  response = call_api(:post, "/suppliers/#{supplier_id}/frameworks/#{framework_slug}", payload: {
```

Comment:
POST or PATCH? As discussed offline, I don't know if we'd ever do this step for a supplier with an existing `frameworkInterest`, with data to overwrite. (Can we even PATCH at the `frameworkInterest` level, like we do for the `declaration`?)

Reply:
> POST or PATCH?

POO. I'm going to have to rethink this one...

It's a weird situation, because even if we do a fetch-modify-post-back it likely won't work without any tinkering due to all the extra properties the serialization annotates onto `User`.

Reply:
Ah - I've looked at the view closer and think we might actually be in the clear: when we have "update" endpoints they tend to do two things - call `SomeModel.update_from_json()` methods to update properties that have straightforward correspondences, but then also special-case clauses to handle some of the more "magic" pseudo-properties. `update_supplier_framework` is _entirely_ special cases. `SupplierFramework` doesn't even have an `update_from_json()` method. Any missing keys on a `POST`ed `frameworkInterest` just get their clauses skipped.

So _that_ will work - it still doesn't save us from the fact that we set `applicationCompanyDetailsConfirmed` with not-actually-complete data... but... it's not as if we don't do that already e.g. `Given 'we accepted that suppliers application to the framework'` for the steps we consider "short cuts".